src/test/java/com/intel/gkl/smithwaterman/SmithWatermanUnitTest.java

```diff
@@ -113,6 +113,33 @@ public void smithWatermanThrowsIllegalArgumentExceptionIfMatchValueGreaterThanMa
     }
 
     @Test(enabled = true)
+    public void singleElementSequencesAlignmentTest(){
+        final IntelSmithWaterman sw = new IntelSmithWaterman();
+        sw.load(null);
+
+        byte[] ref =    new byte [] {'C'};
```

Comment:
Can you please address the size 0 array ref and align in main java class ?

Reply:
A empty array test like: 
 new byte []  {''};

Reply:
Added new checks in SmithWaterman that throw IllegalArgumentException if any of the sequences have a length of 0.

Reply:
Looks to handle in the Java layer.